sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -176,7 +184,9 @@ class StreamExecution(
       // Mark ACTIVE and then post the event. QueryStarted event is synchronously sent to listeners,
       // so must mark this as ACTIVE first.
       state = ACTIVE
-      postEvent(new QueryStarted(this.toInfo)) // Assumption: Does not throw exception.
+      sparkSession.sparkContext.env.metricsSystem.registerSource(streamMetrics)
+      updateStatus()
+      postEvent(new QueryStarted(currentStatus)) // Assumption: Does not throw exception.
```

Comment:
it's good to be consistent. Otherwise if you want to change the behavior of status for example, then this line will not be left orphaned.